fluss-server/src/main/java/com/alibaba/fluss/server/coordinator/LakeTableTieringManager.java

```diff
@@ -0,0 +1,426 @@
+        inLock(
+                lock,
+                () -> {
+                    tablePathById.remove(tableId);
+                    tieringIntervalByTableId.remove(tableId);
+                    tieringStates.remove(tableId);
+                });
+    }
+
+    @Nullable
+    public Tuple2<Long, TablePath> requestTable() {
+        return inLock(
+                lock,
+                () -> {
+                    Long tableId = pendingTieringTables.poll();
+                    // no any pending table, return directly
+                    if (tableId == null) {
+                        return null;
+                    }
+                    TablePath tablePath = tablePathById.get(tableId);
+                    // the table has been dropped, request again
+                    if (tablePath == null) {
+                        return requestTable();
+                    }
+                    registerTieringTimeout(tableId);
+                    doHandleStateChange(tableId, TieringState.Tiering);
+                    return new Tuple2<>(tableId, tablePath);
+                });
+    }
+
+    private void registerTieringTimeout(long tableId) {
+        // we put the table to a timout timer,
+        // if no liveness heartbeat for the table in this period of time, we consider the tiering is
+        // invalid and put it back to Pending state to wait to be requested by other tiering service
+        lakeTieringScheduleTimer.add(
+                new TimerTask(TIERING_SERVICE_TIMEOUT_MS) {
+                    @Override
+                    public void run() {
+                        inLock(
+                                lock,
+                                () -> {
+                                    if (isTieringServiceTimeout(tableId)) {
+                                        // change it to pending to wait to be requested by other
+                                        // lake tiering service
+                                        doHandleStateChange(tableId, TieringState.Pending);
+                                    } else {
+                                        // register the next round of tiering
+                                        // timeout again
+                                        registerTieringTimeout(tableId);
+                                    }
+                                });
+                    }
+                });
+    }
+
+    private boolean isTieringServiceTimeout(long tableId) {
+        Long lastHeartbeat = liveTieringTableIds.get(tableId);
+        return lastHeartbeat == null
+                || clock.milliseconds() - lastHeartbeat >= TIERING_SERVICE_TIMEOUT_MS;
+    }
+
+    public void finishTableTiering(long tableId) {
+        inLock(
+                lock,
+                () ->
+                        // schedule it again if one round of tiering is finished
+                        doHandleStateChange(tableId, TieringState.Scheduled));
```

Comment:
It would be better to add a `Tiered` state, so the state transition becomes `Tiering → Tiered → Scheduled`.
Because transitioning directly from `New/Initialized → Scheduled` is invalid and this can happen in case of a zombie flink job comes back.

Similarity, we should add `Failed` state, to allow the job reports failing and let other jobs pick the table again.